midterm-2023-11-07/code/5104-konstantina-gjurkovic/script.ts

```diff
@@ -0,0 +1,217 @@
+          const sortedOscars = movies.slice().sort((a, b) => oscarsSorter(a, b));
+          displayMovies(sortedOscars);
+      }));
+    for (const movie of movies) {
+      const movieRow = generateMovieRow(movie);
+      container.appendChild(movieRow);
+    }
+  }
+  
+  const generateMovieRow = (movie: MovieData) => {
+    const row = document.createElement("div");
+    row.classList.add("movie-table");
+  
+    const idCell = createMovieCell("ID", movie.id);
+    const titleCell = createMovieCell("Title", movie.title);
+    const directorCell = createMovieCell("Director", movie.director);
+    const yearCell = createMovieCell("Year", movie.year);
+    const genreCell = createMovieCell("Genre", movie.genre.join(", "));
+    const plotCell = createMovieCell("Plot", movie.plot);
+    const castArray = movie.cast.map((castMember) => `${castMember.actor} as ${castMember.character}`);
+    const castCell = createMovieCell("Cast", castArray.join(", "));
+    const oscarsCell = createMovieCell("Oscars", movie.oscars);
+    const ratingCell = createMovieCell("Rating", movie.rating);
+
+    row.appendChild(idCell);
+    row.appendChild(titleCell);
+    row.appendChild(directorCell);
+    row.appendChild(yearCell);
+    row.appendChild(genreCell);
+    row.appendChild(plotCell);
+    row.appendChild(castCell);
+    row.appendChild(oscarsCell);
+    row.appendChild(ratingCell);
+  
+    return row;
+  }
+
+  const createMovieCell = (header: string, content: string | number | object) => {
+    const cell = document.createElement("div");
+    cell.classList.add("movie-data", `movie-${header.toLowerCase()}`);
+  
+    let formattedContent = '';
+  
+    if (typeof content === 'object') {
+      const oscars = content as { [oscarType: string]: string };
+      formattedContent = Object.keys(oscars).map((oscarType) => `${oscarType}: ${oscars[oscarType]}`).join(', ');
+    } else {
+      formattedContent = String(content);
+    }
+  
+    cell.innerHTML = `<strong>${header}:</strong> ${formattedContent}`;
+  
+    return cell;
+  }
+  
+  
+  
+  function filterMoviesByGenre(selectedGenre: string) {
+    if (selectedGenre === "") {
+      displayMovies(movies);
+    } else {
+      const filteredMovies = movies.filter((movie) => {
+        return movie.genre.includes(selectedGenre);
+      });
+      displayMovies(filteredMovies);
+    }
+  }
+  
+  function filterMoviesByActor(actorName: string) {
+    const filteredMovies = movies.filter((movie) => {
+      const lowerActorName = actorName.toLowerCase();
+      return movie.cast.some((castMember) => {
+        const actor = castMember.actor.toLowerCase();
+        const character = castMember.character.toLowerCase();
+        return actor.includes(lowerActorName) || character.includes(lowerActorName);
+      });
+    });
+  
+    displayMovies(filteredMovies);
+  }
+  function filterMoviesByTitle(title: string) {
+    const filteredMovies = movies.filter((movie) => {
```

Comment:
The filter is ok - small nitpick, if we move the `const lowerTitle = title.toLowerCase();` line outside the `filter` method, it will execute only once - now it executes for each and every movie. +10